pkg/registry/core/service/strategy.go

```diff
@@ -464,6 +485,23 @@ func sameHCNodePort(oldSvc, newSvc *api.Service) bool {
 	return oldSvc.Spec.HealthCheckNodePort == newSvc.Spec.HealthCheckNodePort
 }
 
+func needsLoadBalancerClass(svc *api.Service) bool {
+	if svc.Spec.Type != api.ServiceTypeLoadBalancer {
+		return false
+	}
+	if svc.Spec.LoadBalancerClass == nil {
```

Comment:
I find the formulation of these to be hard to think about.

First, "needs" is wrong - nothing NEEDS this field. It's more like "canHave".

Second, I don't thing testing for nil here matter and is distracting.

Third, the "same" func will return true in cases it should not (when one is nil and the other isnt).

Does this work?  It feels more self-documenting:

```
func canSetLoadBalancerClass(svc *api.Service) bool {
￼	return svc.Spec.Type == api.ServiceTypeLoadBalancer
￼}
￼
￼func sameLoadBalancerClass(oldSvc, newSvc *api.Service) bool {
￼	if (oldSvc.Spec.LoadBalancerClass == nil) != (newSvc.Spec.LoadBalancerClass == nil) {
￼		return false
￼	}
    if oldSvc.Spec.LoadBalancerClass == nil {
        return true // both are nil
    }
￼	return *oldSvc.Spec.LoadBalancerClass == *newSvc.Spec.LoadBalancerClass
￼}
```